Test/Unit/Model/Api/ShippingMethodsTest.php

```diff
@@ -0,0 +1,494 @@
+        $this->request = $this->createMock(Request::class);
+        $this->cache = $this->createMock(CacheInterface::class);
+        $this->priceHelper = $this->createMock(PriceHelper::class);
+        $this->sessionHelper = $this->createMock(SessionHelper::class);
+
+        $this->bugsnag = $this->getMockBuilder(Bugsnag::class)
+            ->setMethods(['notifyException'])
+            ->disableOriginalConstructor()
+            ->getMock();
+        $this->bugsnag->method('notifyException')
+            ->willReturnSelf();
+    }
+
+    public function testGetShippingMethodsIfQuoteEmptyReturnException()
+    {
+        $quoteId = 1001;
+        $cart = [
+            'display_id' => '100050001 / '.$quoteId
+        ];
+        $shippingAddress = [
+            'street_address1' => 'test'
+        ];
+
+        $quote = $this->getMockBuilder(Quote::class)
+            ->setMethods(['getId'])
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $quote->method('getId')
+            ->willReturn(false);
+
+        $cartHelper = $this->getMockBuilder(CartHelper::class)
+            ->setMethods(['getQuoteById', 'validateEmail'])
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $cartHelper->method('validateEmail')
+            ->withAnyParameters()
+            ->willReturn(true);
+        $cartHelper->method('getQuoteById')
+            ->willReturn(false);
+
+        $methods = ['sendErrorResponse', 'proceedWithHook', 'checkCartItems', 'getQuoteById', 'notifyException',
+            'validateQuote', 'loadSessionByQuote', 'throwUnknownQuoteIdException', 'catchExceptionAndSendError'
+        ];
+        $this->currentMock = $this->getMockBuilder(BoltShippingMethods::class)
+            ->setMethods($methods)
+            ->setConstructorArgs([
+                $this->hookHelper,
+                $this->regionModel,
+                $this->factoryShippingOptionsMock,
+                $this->shippingTaxInterfaceFactory,
+                $cartHelper,
+                $this->totalsCollector,
+                $this->converter,
+                $this->shippingOptionInterfaceFactory,
+                $this->bugsnag,
+                $this->logHelper,
+                $this->errorResponse,
+                $this->response,
+                $this->configHelper,
+                $this->request,
+                $this->cache,
+                $this->priceHelper,
+                $this->sessionHelper
+            ])
+            ->getMock();
+
+        $message = new Phrase('Unprocessable Entity: Unknown quote id: ' . $quoteId);
+        $this->currentMock->method('throwUnknownQuoteIdException')
+            ->willThrowException(new LocalizedException($message));
+        $this->currentMock->method('catchExceptionAndSendError')
+            ->withAnyParameters()
+            ->willThrowException(new LocalizedException($message));
+
+        try {
```

Comment:
This is not the recommended way of testing exceptions in PHPUnit. You shouldn't have a `try-catch` block - what if your mock somehow doesn't throw an exception?
Let the exception be thrown and assert with either an annotation or `$this->expectException`/`$this->expectExceptionMessage()`.

Reply:
uppated